spec/acceptance/rollback_iteration_spec.rb

```diff
@@ -0,0 +1,172 @@
+require 'spec_helper'
+
+module IterateSpec
+  class AddsOneActionWithRollback
+    extend LightService::Action
+    expects :number
+    promises :number
+
+    executed do |ctx|
+      ctx.number += 1
+    end
+
+    rolled_back do |ctx|
+      ctx.number -= 1
+    end
+  end
+
+  class AddsTwoActionWithRollback
+    extend LightService::Action
+    expects :number
+
+    executed do |context|
+      context.number = context.number + 2
+      context.fail_with_rollback!("I did not like this a bit!")
+    end
+
+    rolled_back do |context|
+      context.number -= 2
+    end
+  end
+
+  class ConcatenateString
+    extend LightService::Action
+    expects :string,
+            :word
+    promises :string
+
+    executed do |context|
+      context.string.concat context.word
+    end
+
+    rolled_back do |context|
+      context.string.chomp! context.word
+    end
+  end
+
+  class ReverseString
+    extend LightService::Action
+    expects :string,
+            :word
+    promises :string
+
+    executed do |context|
+      context.string.reverse!
+    end
+
+    rolled_back do |context|
+      context.string.reverse!
+    end
+  end
+
+  class CallRollback
+    extend LightService::Action
+
+    executed do |context|
+      context.fail_with_rollback!('Arbitrary rollback.')
+    end
+  end
+
+  class CallRollbackIfSpam
```

Comment:
This is awful because this fixture assumes to find the word `spam` which is indeed passed as params by the test...I'll rethink it, but atm it's doing its job.